### PR TITLE
Vector tile layer - part 3 (basic styling)

### PR DESCRIPTION
**Pull request by wonder-sk**
Continued work on vector tile layer implementation.

This adds styling panel integration for configuration of the "basic" renderer: one can set symbol, sub-layer name, min/max zoom levels and filter expression. So far it is pretty rudimentary, without data-defined symbol properties, list of sub-layers, expression builder... but at least something to start with :)

See the QEP: qgis/QGIS-Enhancement-Proposals#162

image(2)

# Thanks

Huge thanks to all funders who have contributed to the crowdfunding and made this possible :clap: :clap: :clap:
https://www.lutraconsulting.co.uk/blog/2020/04/02/vectortiles-donors/

### CODE REVIEW

File: src/gui/vectortile/qgsvectortilebasicrendererwidget.cpp [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+void QgsVectorTileBasicRendererWidget::apply()
+{
+  mVTLayer->setRenderer( mRenderer->clone() );
+}
+
+void QgsVectorTileBasicRendererWidget::addStyle()
+{
+  QStringList lst;
+  lst << tr( "Marker" ) << tr( "Line" ) << tr( "Fill" );
+  QString type = QInputDialog::getItem( this, tr( "Add style" ), tr( "Please choose symbol type" ), lst, 0, false );
```
Comment (nyalldawson) [inline src/gui/vectortile/qgsvectortilebasicrendererwidget.cpp:365]: Would be nicer UI to make btnAddRule a QToolButton and put this choice in a popup menu

**Comment by nyalldawson** [comment]
LGTM!

File: src/gui/vectortile/qgsvectortilebasicrendererwidget.cpp (hunk 1 above)
Comment (wonder-sk) [reply src/gui/vectortile/qgsvectortilebasicrendererwidget.cpp]: Good point - updated!

**Comment by NathanW2** [comment]
YES!

**Pull request merged**